### ISSUE TITLE
supports for flutter 3.10.0 

### ISSUE THREAD

**Issue by Nobler** [tjarvstrand/flutter_timezone#15]
flutter 3.10.0 requires the kotlin version at least 1.6.20

```
* What went wrong:
The Android Gradle plugin supports only kotlin-android-extensions Gradle plugin version 1.6.20 and higher.
The following dependencies do not satisfy the required version:
project ':flutter_timezone' -> org.jetbrains.kotlin:kotlin-gradle-plugin:1.6.10
```

**Comment by tjarvstrand** [comment]
Hi @Nobler, I'm using 3.10 myself with no issue. Did you perhaps upgrade your Android Gradle plugin?

I'll take a look at this but in the meantime, you might be able to work around the issue by downgrading AGP.

**Comment by tjarvstrand** [comment]
Which versions of Android, AGP, Kotlin and kotlin-gradle-plugin are you on?

**Comment by Nobler** [comment]
Thanks for the replying.
I don't build a apk but a aar, so configurations is from flutter SDK (I don't 100% sure):
1. ext.kotlin_version = '1.7.10'
2. classpath 'com.android.tools.build:gradle:7.3.0'
3. compileSdkVersion 31

**Comment by tjarvstrand** [comment]
I've now upgraded my own project to use AGP 7.4.2, Gradle 7.5.0 and Kotlin 1.9.0 and I can't reproduce this issue.

What is your version of the Android Gradle plugin?

Do you have anything specifying the version of the Kotlin Gradle plugin in your `build.gradle`? E.g. `classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"`

**Comment by Nobler** [comment]
@tjarvstrand Hi, sorry for misrepresenting the issue. I finally find that it is because some other plugins use `kotlin-android-extensions` that requiring kotlin version at least 1.6.20.
You can reproduct the issue by adding this line in `android/build.gradle`:
```diff
diff --git a/android/build.gradle b/android/build.gradle
index 29278e3..b37aadc 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -23,6 +23,7 @@ rootProject.allprojects {
 
 apply plugin: 'com.android.library'
 apply plugin: 'kotlin-android'
+apply plugin: 'kotlin-android-extensions'
 
 android {
     namespace 'net.wolverinebeach.flutter_timezone'
```
I will try to solve the issue by manipulating those plugins.
You can also consider upgrading the kotlin version, for helping others that facing the same issue.

**Comment by btrautmann** [comment]
I'm seeing the same using the following versions:
```
AGP: 8.1.2
Kotlin: 1.7.21
```
Error produced:
```console
FAILURE: Build failed with an exception.

* What went wrong:
The Android Gradle plugin supports only kotlin-android-extensions Gradle plugin version 1.6.20 and higher.
The following dependencies do not satisfy the required version:
project ':flutter_timezone' -> org.jetbrains.kotlin:kotlin-gradle-plugin:1.6.10
```
I've gotten around this with internal packages by bumping the kotlin-gradle-plugin version to the indicated version (1.6.20 in this case)

**Comment by Nobler** [comment]
> I'm seeing the same using the following versions:
> 
> ```
> Gradle: 8.3
> AGP: 8.1.2
> Kotlin: 1.7.21
> ```
> 
> Error produced:
> 
> ```
> FAILURE: Build failed with an exception.
> 
> * What went wrong:
> The Android Gradle plugin supports only kotlin-android-extensions Gradle plugin version 1.6.20 and higher.
> The following dependencies do not satisfy the required version:
> project ':flutter_timezone' -> org.jetbrains.kotlin:kotlin-gradle-plugin:1.6.10
> ```
> 
> I've gotten around this with internal packages by bumping the kotlin-gradle-plugin version to the indicated version (1.6.20 in this case)

You can migration `kotlin-android-extensions`, see https://stackoverflow.com/a/65179769/4083396

**Comment by btrautmann** [comment]
> You can migration `kotlin-android-extensions`, see https://stackoverflow.com/a/65179769/4083396

Sure, I understand that--our project is not in a position to do so, nor do we want to invest further in our native code. Migration won't always be an option for folks, and the change on the package-side is pretty simple.

**Comment by Nobler** [comment]
> > You can migration `kotlin-android-extensions`, see https://stackoverflow.com/a/65179769/4083396
> 
> Sure, I understand that--our project is not in a position to do so, nor do we want to invest further in our native code. Migration won't always be an option for folks, and the change on the package-side is pretty simple.

Yes. I also recommend to upgrade the kotlin version. But in another words, the deprecated k-a-e plugin should be removed as well.

**Comment by tjarvstrand** [comment]
Apologies for dropping the ball a bit on this. I've been pretty busy lately. Just released a new version, bumping the Kotlin version to 1.6.21, which should solve the issue.

Please let me know if you still have problems with version 1.0.8.

**Comment by Nobler** [comment]
Add a note for anyone including me having an interest in the root cause.

1. The error was prompted by AGP starting from version 7.3. see https://issuetracker.google.com/issues/215545075
2. Flutter 3.7 upgraded the AGP to 7.3. see [const String templateAndroidGradlePluginVersion = '7.3.0';](https://github.com/flutter/flutter/blame/bb856079c40139fc732927c73f1e882d77234dc5/packages/flutter_tools/lib/src/android/gradle_utils.dart#L31)
3. Our project upgraded Flutter from 3.3 to 3.10 with the use of `kotlin-android-extensions`, so the error occurred.

Also, the best solution is to migrate `kotlin-android-extensions` because it will be removed in Kotlin 1.8 (see https://android-developers.googleblog.com/2020/11/the-future-of-kotlin-android-extensions.html).
